mesh/mesh.cpp

```diff
@@ -2774,6 +2774,41 @@ void Mesh::DoNodeReorder(DSTable *old_v_to_v, Table *old_elem_vert)
    Nodes->Update(); // just needed to update Nodes->sequence
 }
 
+void Mesh::SetPatchAttribute(int i, int attr)
+{
+   MFEM_ASSERT(NURBSext, "SetPatchAttribute is only for NURBS meshes");
+   NURBSext->SetPatchAttribute(i, attr);
+   const Array<int>& elems = NURBSext->GetPatchElements(i);
+   for (auto e : elems)
```

Comment:
This kind of for loop with `for ( auto i: array)` might not be very consistent with a large part of the code in mesh.cpp. However, I see some applications in mesh.cpp and it forms a good alternative here, especially with the reference on line 2781.

Reply:
This type of loop was introduced in C++11, but much of MFEM was written before C++11 was introduced in MFEM.